chain/chain/src/store.rs

```diff
@@ -216,15 +216,21 @@ pub trait ChainStoreAccess {
     /// so the receipts from these blocks are propagated
     fn get_incoming_receipts_for_shard(
         &self,
-        shard_id: ShardId,
+        epoch_manager: &dyn EpochManagerAdapter,
+        mut shard_id: ShardId,
         mut block_hash: CryptoHash,
         last_chunk_height_included: BlockHeight,
     ) -> Result<Vec<ReceiptProofResponse>, Error> {
+        let _span =
+            tracing::debug_span!(target: "store", "get_incoming_receipts_for_shard").entered();
+
         let mut ret = vec![];
+        let mut header = self.get_block_header(&block_hash)?;
 
-        loop {
-            let header = self.get_block_header(&block_hash)?;
+        let target_shard_id = shard_id;
+        let target_shard_layout = epoch_manager.get_shard_layout_from_prev_block(&block_hash)?;
 
+        loop {
```

Comment:
implementation nit:
instead of maintaining mutable `header` variable you can just do the following:
```
loop {
    let header =  self.get_block_header(&block_hash)?;
   ...
}
```